packages/core/src/components/overlay/overlay.tsx

```diff
@@ -404,9 +419,80 @@ export class Overlay extends AbstractPureComponent2<OverlayProps, IOverlayState>
         }
     }
 
+    private renderDummyElement(ref: (element: HTMLDivElement) => void, key: string) {
+        const { transitionDuration, transitionName } = this.props;
+        return (
+            <CSSTransition
+                classNames={transitionName}
+                key={key}
+                addEndListener={this.handleTransitionAddEnd}
+                timeout={transitionDuration}
+                unmountOnExit={true}
+            >
+                <div ref={ref} tabIndex={0} />
+            </CSSTransition>
+        );
+    }
+
+    private handleStartFocusTrapElementFocusIn = (e: FocusEvent) => {
+        e.preventDefault();
+        e.stopImmediatePropagation();
+        if (
+            e.relatedTarget != null &&
+            this.containerElement!.contains(e.relatedTarget as Element) &&
+            e.relatedTarget !== this.endFocusTrapElement
+        ) {
+            this.endFocusTrapElement?.focus();
+        } else {
+            this.keyboardFocusableElements().shift()?.focus();
+        }
+    };
+
+    private handleEndFocusTrapElementFocusIn = (e: FocusEvent) => {
+        e.preventDefault();
+        e.stopImmediatePropagation();
+        if (
+            e.relatedTarget != null &&
+            this.containerElement!.contains(e.relatedTarget as Element) &&
+            e.relatedTarget !== this.startFocusTrapElement
+        ) {
+            this.startFocusTrapElement?.focus();
+        } else {
+            this.keyboardFocusableElements().pop()?.focus();
+        }
+    };
+
+    private keyboardFocusableElements() {
+        const focusableElements: HTMLElement[] =
+            this.containerElement !== null
+                ? Array.from(
+                      // Order may not be correct if children elements use tabindex values > 0
+                      this.containerElement.querySelectorAll(
+                          [
```

Comment:
I'm curious how you decided on this list? Is there some documentation we can link to? It would be nice to leave some more comments about our reasoning for this code

Reply:
I cobbled this together from jquery-ui's [`focusable` selector](https://github.com/jquery/jquery-ui/blob/70dae67b73dfea9126f126f516fe8286f1e73417/ui/focusable.js#L45) and this [SO answer](https://stackoverflow.com/a/1600194/2213364). I will add a comment